daemon/containerd/image_manifest.go

```diff
@@ -223,12 +223,18 @@ func (m *ImageManifest) ImagePlatform(ctx context.Context) (ocispec.Platform, er
 		return *target.Platform, nil
 	}
 
+	var out ocispec.Platform
+	err := m.ReadConfig(ctx, &out)
+	return out, err
+}
+
+// ReadConfig gets the image config and unmarshals it into the provided struct.
+// The provided struct should be a pointer to the config struct or its subset.
+func (m *ImageManifest) ReadConfig(ctx context.Context, outConfig any) error {
```

Comment:
Adding `any` may require this file to have a `go:build` comment (https://github.com/moby/moby/pull/46941) (although I don't think this file would be imported by consumers)


Looks like this is only used internally; so we may not need to export it; more so because having to pass the type to read feels a bit "internal"; i.e., if it's encapsulated in the `ImageManifest` type, then somewhat expecting that to know what it returns; being able to unmarshal to something else is mostly an internal optimisation.

Reply:
I can change it to `interface{}`.

This struct already has most of the methods marked as exported (to signify that they're not "internal" and should be used by the consumers of the `ImageManifest`).

I think we could just unexport the whole `ImageManifest` instead.. WDYT?

Reply:
Unexporting the method itself is more of an indicator that the method is "internal" to the `ImageManifest` itself though, which is not the case here as it's used outside of the struct itself by the `ImageService`.

Reply:
Maybe. It's just that an interface could still be matched if it would ever be returned, so doesn't really remain internal; e.g. the `errPlatformNotFound` is not exported, but still satisfies the `errdefs.NotFound` interface.

It shouldn't be _really_ problematic (just don't do that!), just thinking if un-exporting it makes it even more clear and it for sure can help looking at code, _knowing_ that it cannot be used outside of the package, without having to go search if it _possibly_ was exported to match some interface .. somewhere..